src/api/app/models/workflow.rb

```diff
@@ -31,11 +31,17 @@ def call
       restore_target_projects
     when scm_webhook.new_pull_request?, scm_webhook.updated_pull_request?, scm_webhook.push_event?, scm_webhook.tag_push_event?
       steps.each do |step|
-        step.call({ workflow_filters: filters })
+        # TODO: improve this call.
```

Comment:
What should be improved exactly? Will it happen in this PR?

Reply:
Sorry, I forgot to remove the comment (on it right now). It was added before I moved the code to the method `call_step_and_collect_artifacts` because of high cyclomatic complexity.

Reply:
comment removed